docs/source/app-configuration.rst

```diff
@@ -0,0 +1,160 @@
+
+
+Getting Started
+---------------
+
+An initial deploy of StreamAlert must be performed before StreamAlert Apps can be configured. If you do not have a current deploy,
+please visit the `Project's Getting Started <getting-started.html>`_ page to get up and running.
+
+
+Adding a new StreamAlert App requires the following steps:
+
+1. Configure the StreamAlert App through the CLI (via ``python manage.py app new``).
+2. Enter the required authentication information for the app being configured.
+3. Deploy the new App and the Rule Processor to accept incoming data from your App.
+
+To get help configuring a new StreamAlert App, use:
+
+.. code-block:: bash
+
+  $ python manage.py app new --help
+
+
+1. Configure the StreamAlert App
+````````````````````````````````
+
+The StreamAlert CLI is used to add a new App configuration.
+
+.. code-block:: bash
+
+  $ python manage.py app new \
+  --type duo_auth \
+  --cluster prod \
+  --name duo_prod_collector \
+  --interval 'rate(2 hours)' \
+  --timeout 80 \
+  --memory 128
+
+
+.. note:: Duo Security's Admin API is limited to two (2) requests per-minute. Therefore, setting the ``--timeout`` flag to any value between 10 and 60 will be of no additional value. A recommended timeout of 80 seconds will guarantee four (4) requests happen per-execution.
+
+
+
+=========================  ===========
+Flag                       Description
+-------------------------  -----------
+``--type``                 Type of app integration function being configured. Current choices are: `duo_auth`, `duo_admin`
+``--cluster``              Applicable cluster this function should be configured against.
+``--name``                 Unique name to be assigned to this app integration function. This is useful when configuring multiple accounts per service.
+``--interval``             The interval, defined using a 'rate' expression, at which this app integration function should execute. See AWS Schedule `Rate Expressions <http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html#RateExpressions>`_.
+``--timeout``              The AWS Lambda function timeout value, in seconds. This should be an integer between 10 and 300.
+``--memory``               The AWS Lambda function max memory value, in megabytes. This should be an integer between 128 and 1536.
+=========================  ===========
+
+
+
+2. Enter the required authentication information
+````````````````````````````````````````````````
+
+The above command will result in a few prompts asking for the required authentication information needed to configure this StreamAlert App.
+
+.. note:: After the last required authentication value is entered, the values are sent to AWS SSM's `Parameter Store <https://aws.amazon.com/ec2/systems-manager/parameter-store/>`_ as a ``SecureString`` to be used as part of this App's config. Due to this requirement, please ensure you have the correct and valid AWS credentials loaded before continuing.
```

Comment:
This seems like an unnecessary comment - im assuming the code will let them know if their creds are invalid?

Reply:
It will - but this comment exists more so to highlight this line:

> please ensure you have the _correct_ ..

My reasoning is that users are going to be entering sensitive data into the CLI here, and since the whole uploading to AWS happens very seamlessly, they could inadvertently end up pushing sensitive creds to an AWS account they don't mean to push them to..

Thoughts?

Reply:
Oh I see. Perhaps we can prompt the user during deploy, asking if they're sure they want to deploy the lambda function to account ID 1234?

Reply:
Not a bad suggestion - this all actually happens during configuration (prior to even deploying). Not saying that logic couldn't occur during configuration also - will need to make a note of it or file an issue.